src/main/java/handlers/PartialContentHandler.java

```diff
@@ -0,0 +1,127 @@
+package handlers;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.util.Arrays;
+import java.util.regex.Pattern;
+import messages.HTTPStatus;
+import messages.Request;
+import messages.Response;
+
+public class PartialContentHandler extends FileSystemHandler {
+
+  private int contentLength;
+  private File file;
+  private byte[] fullContent;
+
+  enum Range {
+    FULL_RANGE {
+      @Override
+      boolean canBeParsedToNumber(String rangeValues) {
+        return Pattern.matches("\\d+-\\d+", rangeValues);
+      }
+    },
+    VALUE_TO_END {
+      @Override
+      boolean canBeParsedToNumber(String rangeValues) {
+        return Pattern.matches("-\\d+", rangeValues);
+      }
+    },
+    VALUE_FROM_END {
+      @Override
+      boolean canBeParsedToNumber(String rangeValues) {
+        return Pattern.matches("\\d+-", rangeValues);
+      }
+    };
+    abstract boolean canBeParsedToNumber(String rangeValues);
+  }
```

Comment:
I like this idea (overriding the number-format-checking function in three different ways for the different range types). You're on your way to using the strategy pattern to clean this class up, but I think you should take it a little farther and introduce some inheritance and polymorphism to make this more readable (see my other comment for more info).